theano/gpuarray/c_code/dnn_fwd.c

```diff
@@ -358,53 +395,72 @@ APPLY_SPECIFIC(conv_fwd)(PyGpuArrayObject *input, PyGpuArrayObject *kerns,
             use_cached ? "(cache)": "",
             worksize,
             hashkey.c_str()
-      );
+    );
+    if (!(reuse_algo || use_cached)) {
+        // We have selected an algorithm at runtime.
+        // `t` still contains timing about selection step.
+        fprintf(stderr, "\t(selected %s fwd algo in %g milliseconds)\n", selection_name, theano_clock_to_milliseconds(t));
+        if (n_selections > 1) {
+            fprintf(stderr, "\t(selected %lu fwd algos in %g milliseconds (average: %g milliseconds per selection))\n",
+                    n_selections,
+                    theano_clock_to_milliseconds(total_selection_time),
+                    theano_clock_average_to_milliseconds(total_selection_time, n_selections));
+        }
+    }
   }
 #endif
 
   if (params->choose_once) {
     reuse_algo = 1;
   }
 
-  {
-    gpudata *workspace = 0;
-    /*
-     * This is less than ideal since we need to free it after (which
-     * introduces a synchronization point. But we don't have a module
-     * to place a nice get_work_mem() function in.
-     */
-    if (worksize != 0) {
-      workspace = gpudata_alloc(c->ctx, worksize, NULL, 0, NULL);
-      if (workspace == NULL) {
-        PyErr_SetString(PyExc_RuntimeError, "Could not allocate working memory");
-        cuda_exit(c->ctx);
-        return 1;
-      }
+  gpudata *workspace = 0;
+  /*
+   * This is less than ideal since we need to free it after (which
+   * introduces a synchronization point. But we don't have a module
+   * to place a nice get_work_mem() function in.
```

Comment:
@notoraptor do you understand this comment?

We cache by default the call to cudaMalloc, so it don't create a real sync. It could create a sync if we where computing many ops in parallel, but currently we put everything on one stream. So this don't currently create any sync point.

@borisfom is it you that wrote that comment?